patches/server/0575-Configurable-door-breaking-difficulty.patch

```diff
@@ -5,58 +5,65 @@ Subject: [PATCH] Configurable door breaking difficulty
++                    java.util.Arrays.stream(net.minecraft.world.Difficulty.values())
++                        .filter(difficultyPredicate)
++                        .collect(Collectors.toList()),
++                    net.minecraft.world.Difficulty.class
++                )
++            );
++        }
 +    }
 +
      public short keepLoadedRange;
      private void keepLoadedRange() {
          keepLoadedRange = (short) (getInt("keep-spawn-loaded-range", Math.min(spigotConfig.viewDistance, 10)) * 16);
 diff --git a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
-index ddf77350b1762be523eeed760669bc33f634109f..85294ab8eaa654be9358f489cec7e1ae665fc34b 100644
+index ddf77350b1762be523eeed760669bc33f634109f..d8bd9be2a7d2ce5cbbf39373cea1d1d9294c5354 100644
 --- a/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
 +++ b/src/main/java/net/minecraft/world/entity/monster/Vindicator.java
 @@ -195,7 +195,7 @@ public class Vindicator extends AbstractIllager {
  
      static class VindicatorBreakDoorGoal extends BreakDoorGoal {
          public VindicatorBreakDoorGoal(Mob mob) {
 -            super(mob, 6, Vindicator.DOOR_BREAKING_PREDICATE);
-+            super(mob, 6, com.google.common.base.Predicates.in(mob.level.paperConfig.vindicatorBreakDoors)); // Paper
++            super(mob, 6, com.google.common.base.Predicates.in(mob.level.paperConfig.entitiesDifficultyBreakDoors.get(mob.getType()))); // Paper
              this.setFlags(EnumSet.of(Goal.Flag.MOVE));
          }
  
 diff --git a/src/main/java/net/minecraft/world/entity/monster/Zombie.java b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
-index 64222cced10e87438ee3ada26aead0284649be78..e72e9b748b3f3e34baddf01366c703efba50c67c 100644
+index 64222cced10e87438ee3ada26aead0284649be78..447d580d812bb091e5119fdc706a3de8bc07e386 100644
 --- a/src/main/java/net/minecraft/world/entity/monster/Zombie.java
 +++ b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
-@@ -97,7 +97,7 @@ public class Zombie extends Monster {
+@@ -97,7 +97,13 @@ public class Zombie extends Monster {
  
      public Zombie(EntityType<? extends Zombie> type, Level world) {
          super(type, world);
 -        this.breakDoorGoal = new BreakDoorGoal(this, Zombie.DOOR_BREAKING_PREDICATE);
-+        this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.zombieBreakDoors)); // Paper
++        // Paper start - Adjust breakDoorGoal based in entity
++        if (world.paperConfig.entitiesDifficultyBreakDoors.containsKey(type)) {
++            this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.entitiesDifficultyBreakDoors.get(type)));
++        } else {
++            this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.entitiesDifficultyBreakDoors.get(EntityType.ZOMBIE)));
```

Comment:
in my mind this only can happen if for any reason a user edit the EntityType for a custom mob of any in a fork....